plugin/cmake.vim

```diff
@@ -11,6 +11,7 @@ let g:cmake_configure_arguments = get(g:, 'cmake_configure_arguments', '-D CMAKE
 let g:cmake_build_arguments = get(g:, 'cmake_build_arguments', '')
 let g:cmake_asyncrun_options = get(g:, 'cmake_asyncrun_options', {'save': 2})
 let g:cmake_target_asyncrun_options = get(g:, 'cmake_target_asyncrun_options', {})
+let g:cmake_dap_configuration  = get(g:, 'cmake_dap_configuration', {})
```

Comment:
Let's move `local config` definitions here to avoid creating and extending the table every time:

```suggestion
let g:cmake_dap_configuration  = get(g:, 'cmake_dap_configuration', {
    type: 'cpp',
    name: 'Debug CMake target',
    request: 'launch',
  })
```

And override target, cwd and working directory at run.

Reply:
OK.

Reply:
`args` and `program` are dynamically created, we could not avoid creating the table every time, if we provide some default key-value pairs, it means we need to extend and override the table every time even if we do not customize this variable in user settings. I still prefer to using the empty table as the default value of g:cmake_dap_configuration.

Any idea?